src/mongocrypt-endpoint.c

```diff
@@ -55,6 +56,7 @@ _mongocrypt_endpoint_new (const char *endpoint_raw,
    char *host_end;
 
    endpoint = bson_malloc0 (sizeof (_mongocrypt_endpoint_t));
```

Comment:
Ok, bson_malloc0() happily calls abort() on failure, no need to check it.

Reply:
Is this legal C89 syntax? I vaguely recall having to write: "sizeof struct _type", but that was many an age ago!

Reply:
> Is this legal C89 syntax? I vaguely recall having to write: "sizeof struct _type", but that was many an age ago!

AFAIK it is legal C89 based on an experiment compiling with `-std=c89`: https://godbolt.org/z/9jhM65oYW

Though libmongocrypt is not compiled with `-std=c89` in Evergreen.

FWIW the C89-isms in libmongocrypt, like avoiding mixing declarations and code, is likely attributable to my own habits rather than a requirement. Even then, it is not consistent. Here is one example of mixed declaration and code: https://github.com/mongodb/libmongocrypt/blob/8c9ed66d45ac0b2a33d3b5d3fa533e52bde9509e/src/mongocrypt-buffer.c#L449:L449